src/expand.ts

```diff
@@ -22,14 +22,20 @@ const log = createLogger('expand');
  * references will be available when performing an expansion. Those guaranteed
  * variables are specified as properties on this interface.
  */
-export interface RequiredExpansionContextVars {
-  buildKit: string;
-  buildType: string;
+interface RequiredExpansionContextVars {
+  [key: string]: string;
```

Comment:
It seems odd to put this in the "required" section since it's unbounded.  I think it makes more sense to put it in the derived classes even though it means duplicating it for each.